project/profiles/capacity4more/modules/c4m/og/c4m_og/c4m_og.module

```diff
@@ -108,6 +108,205 @@ function c4m_og_form_alter(&$form, $form_state) {
+
+  $admins = array();
+  // Setting the list of the group admins.
+  foreach(c4m_og_get_group_admins($etid) as $admin) {
+    $admins[$admin->uid] = c4m_user_name($admin);
+  }
+
+  $form['new_owner'] = array(
+    '#type' => 'select',
+    '#options' => $admins,
+    '#title' => t('Select new owner'),
+    '#description' => t('Select the name of the member you wish to be the new owner.'),
+  );
+
+  $form['group'] = array(
+    '#type' => 'value',
+    '#value' => $etid,
+  );
+
+  $form['type'] = array(
+    '#type' => 'value',
+    '#value' => $entity_type,
+  );
+
+  $form['submit'] = array(
+    '#type' => 'submit',
+    '#value' => t('Replace'),
+  );
+
+  return $form;
+}
+
+/**
+ * Submit handler;
+ *
+ * Replacing the owner of a group.
+ */
+function c4m_og_replace_owner_build_form_submit($form, &$form_state) {
+  $new_owner = array(
+    'uid' => $form_state['values']['new_owner'],
+    'name' => $form['new_owner']['#options'][$form_state['values']['new_owner']],
+  );
+  c4m_og_replace_owner($form_state['values']['type'], $form_state['values']['group'], $new_owner);
+
+  // Redirect back to the group page.
+  $form_state['redirect'] = $form_state['values']['type'] . '/' . $form_state['values']['group'] . '/group';
+}
+
+/**
+ * Validation function.
+ */
+function c4m_og_replace_owner_build_form_validate($form, &$form_state) {
+  if (!$form_state['values']['new_owner']) {
+    form_set_error('c4m_og_replace_owner', t('Must select a new owner from the list.'));
+  }
+}
+
+/**
+ * Replacing the owner of a group.
+ *
+ * After replacing the owner of the group we grant "group administrator member"
+ * role to the old owner and revoke this role from the new one.
+ *
+ * @param string $group_type
+ *   The type of the group.
+ * @param int $gid
+ *   The ID of the group.
+ * @param array $new_owner
+ *   An array containing the ID and the name of the new owner.
+ */
+function c4m_og_replace_owner($group_type, $gid, $new_owner) {
+  $group = entity_load_single($group_type, $gid);
+  $old_owner_id = $group->uid;
+
+  // Update the uid on the entity.
+  $group->uid = $new_owner['uid'];
+
+  // Save the entity.
+  if (entity_save($group_type, $group)) {
+    drupal_set_message(t('Failed to change manager to @user.', array('@user' => $new_owner['name'])), 'error');
+  }
+  else {
+    drupal_set_message(t('Owner changed to @user.', array('@user' => $new_owner['name'])));
+
+    $group_admin_role = c4m_og_user_role_load_by_name('administrator member', 'group');
+    // Grant admin role to the old owner.
+    og_role_grant($group_type, $gid, $old_owner_id, $group_admin_role->rid);
+    // Revoke admin role from the new owner.
+    og_role_revoke($group_type, $gid, $new_owner['uid'], $group_admin_role->rid);
+  }
+}
+
+/**
+ * Access callback for the "Replace owner" form.
+ *
+ * @param $group_type
+ * @param $gid
+ * @return bool
+ */
+function c4m_og_user_is_owner($group_type, $gid) {
```

Comment:
btw, since we know the group type is always node, it might be ok to remove the `$group_type` from the signature.